### ISSUE TITLE
Nesting of tags

### ISSUE THREAD

**Issue by JamborJan** [twostairs/paperwork#282]
It would be a huge organizational improvement to have the possibility to nest tags. 

For example when I want to create groups of tags for when or where to do tasks or with whom or which project, I create parent tags and would nest the child tags underneath that.

Thats how it would look like in a very simple mockup:
![bildschirmfoto_2015-03-15_um_14_18_36](https://cloud.githubusercontent.com/assets/7508960/6656154/de0d00ee-cb1e-11e4-8948-e3f85d21a203.png)

**Comment by clemlaf** [comment]
I'm working on the implementation of this feature. Some info about design options I made:
- only one level of nesting is allowed,
- the tag icon is replaced by the chevron when a tag has children,
- the nesting is handled by drag and drop. To move back a leaf tag the root an "unlink" button is available in the "manage tags" box.

However I have a difficulty with angular and bootstrap. 
The sidebar is a tree, and my intention is to add a level to that tree. On the sidebar tree, the collapsing/expanding of the branches is handled by a bootstrap jQuery function. However the Tags and Notebooks branches are populated by the angular controller after boostrap has made the tree "collapsable". If I pause the execution of bootstrap.min.js, and I wait for the tree to be populated, then start again bootstrap.min.js, I have what I want, which is an expandle/collapsable tags tree. 
I would need help to understand better how the execution of the 2 codes can be more harmoniously controlled.
![nesttag1](https://cloud.githubusercontent.com/assets/11900571/8317341/fca48310-19fc-11e5-9b7d-7864ac41e2e8.png)
![nesttag2](https://cloud.githubusercontent.com/assets/11900571/8317342/fca4c942-19fc-11e5-98c7-20bca8e8cfc4.png)

**Comment by Baldrs** [comment]
Please remember, that jQuery is EVIL, especially in Angular code, so I'm going to remove all the code that uses jQuery and rewrite it Angular way when I have enough time to start. For example, I propose to replace all bootstrap javascript with this: https://angular-ui.github.io/bootstrap/. You can use it to rewrite your problematic components too.

So if you can help to start this process, it is appreciated.

**Comment by clemlaf** [comment]
Ok, I solved the problem using angular-ui-bootstrap. I tried to look into the [abn-tree](https://github.com/nickperkinslondon/angular-bootstrap-nav-tree) directive, but it does not seem to support drag and drop from and onto the leaves of the tree so I kept with the structure of sidebar tree as it is.

Also, I replaced the call to the bootstrap-tree script with the collapse directive of angular-ui-bootstrap for all of the sidebar tree (tree headers such as Shortcuts, Notebooks, Tags, as well as the Tags subtree).

I'm sending a pull request with this.
